lib/msf/core/modules/external/bridge.rb

```diff
@@ -197,8 +197,19 @@ def self.applies?(module_name)
 
   def initialize(module_path, framework: nil)
     super
-    gopath = ENV['GOPATH'] || ''
-    self.env = self.env.merge({ 'GOPATH' => File.expand_path('../go', __FILE__) + File::PATH_SEPARATOR + gopath})
+    default_go_path = ENV['GOPATH'] || ''
+    shared_module_lib_path = File.dirname(module_path) + "/shared"
+    go_path = File.expand_path('../go', __FILE__)
+
+    if File.exist?(default_go_path)
+      go_path = go_path + File::PATH_SEPARATOR + default_go_path
+    end
+
+    if File.exist?(shared_module_lib_path)
+      go_path = go_path + File::PATH_SEPARATOR + shared_module_lib_path
```

Comment:
Consider using `File.join` for this:
```
go_path = File.join(go_path, shared_module_lib_path)
```

Reply:
don't know if that does what I want - note PATH_SEPARATOR

Reply:
My mistake, I overlooked that this is for env.